src/testStatusCodeLensProvider.ts

```diff
@@ -0,0 +1,78 @@
+"use strict";
+import { CancellationToken, CodeLens, CodeLensProvider, commands, Disposable, Event, EventEmitter, Range, SymbolInformation, SymbolKind, TextDocument, workspace } from "vscode";
+import { TestResult } from "./testResult";
+import { TestResultsFile } from "./testResultsFile";
+import { TestStatusCodeLens } from "./testStatusCodeLens";
+
+export class TestStatusCodeLensProvider implements CodeLensProvider {
+    private disposables: Disposable[] = [];
+    private enabled: boolean;
+    private onDidChangeCodeLensesEmitter = new EventEmitter<void>();
+
+    // Store everything in a map so we can remember old tests results for the
+    // scenario where a single test is ran. If the test no longer exists in
+    // code it will never be mapped to the symbol, so no harm (though there is
+    // a memory impact)
+    private testResults = new Map<string, TestResult>();
+
+    public constructor(testResultFile: TestResultsFile) {
+        this.checkEnabledOption();
+
+        this.disposables.push(
+            testResultFile.onNewResults(this.addTestResults, this));
+
+        this.disposables.push(
+            workspace.onDidChangeConfiguration(this.checkEnabledOption, this));
+    }
+
+    public dispose() {
+        while (this.disposables.length) {
+            this.disposables.pop().dispose();
+        }
+    }
+
+    public get onDidChangeCodeLenses(): Event<void> {
+        return this.onDidChangeCodeLensesEmitter.event;
+    }
+
+    public provideCodeLenses(document: TextDocument, token: CancellationToken): CodeLens[] | Thenable<CodeLens[]> {
```

Comment:
Try with xUnit, NUnit and MSTest. Seems only xUnit is showing the test result icon. It would be great if you could also support NUnit and MSTest. 😃

Reply:
This should now be fixed in the latest version, I've tried xUnit, MSTest and NUnit (the later two have the same format)